src/Listeners/DuplicateSubscriptionHandler.php

```diff
@@ -80,117 +80,96 @@ public function cancelAndExtendDuplicateSubscriptions($userId)
+                    $this->ecommerceEntityManager->flush($userSubscription);
 
-                        event(new SubscriptionUpdated($oldUserSubscription, $userSubscription));
-                    }
+                    event(new SubscriptionUpdated($oldUserSubscription, $userSubscription));
                 }
-
-            } else {
-                // otherwise, find all duplicate subscriptions with an expiration date in the future,
-                // add up their remaining time, then set the latest active one with that expiration date
-                $totalMinutes = 0;
-
-                /**
-                 * @var $mostRecentlyPurchasedActiveSubscription Subscription|null
-                 */
-                $mostRecentlyPurchasedActiveSubscription = null;
-
-                // extend the date of the most recently created subscription
-                foreach ($allUsersSubscriptions as $userSubscription) {
-
-                    if (!empty($userSubscription->getProduct()) &&
-                        $userSubscription->getProduct()->getBrand() == $brand &&
-                        in_array($userSubscription->getProduct()->getSku(), $membershipProductSkus) &&
-                        $userSubscription->getPaidUntil() > Carbon::now() &&
-                        $userSubscription->getIsActive()) {
-
-                        $totalMinutes += Carbon::now()->diffInMinutes($userSubscription->getPaidUntil());
-
-                        if ($userSubscription->getIsActive() &&
-                            empty($userSubscription->getCanceledOn())) {
-
-                            if (empty($mostRecentlyPurchasedActiveSubscription) ||
-                                $mostRecentlyPurchasedActiveSubscription->getCreatedAt() < $userSubscription->getCreatedAt()) {
-                                $mostRecentlyPurchasedActiveSubscription = $userSubscription;
-                            }
+            }
+        } else {
+            // otherwise, find all duplicate subscriptions with an expiration date in the future,
+            // add up their remaining time, then set the latest active one with that expiration date
+            $totalMinutes = 0;
+
+            /**
+             * @var $mostRecentlyPurchasedActiveSubscription Subscription|null
+             */
+            $mostRecentlyPurchasedActiveSubscription = null;
+
+            // extend the date of the most recently created subscription
+            foreach ($allUsersSubscriptions as $userSubscription) {
+                if (!empty($userSubscription->getProduct()) &&
+                    $userProduct->getProduct()->isMembershipProduct() &&
+                    $userSubscription->getPaidUntil() > Carbon::now() &&
+                    $userSubscription->getIsActive()) {
+                    $totalMinutes += Carbon::now()->diffInMinutes($userSubscription->getPaidUntil());
+
+                    if ($userSubscription->getIsActive() &&
```

Comment:
Active would weed out the canceled subscriptions, so it wouldn't keep adding the time on a subsequent subscription purchase.

The empty($userSubscription->getCanceledOn()) is for selecting the most recent active subscription

Reply:
Ah thanks for clarifying the `empty($userSubscription->getCanceledOn())`. My comment about `$userSubscription->getIsActive()` being redundant is just that the same check for `$userSubscription->getIsActive()` is on line 127 in the `if` statement above, so that's always true already if it gets to line 130.